### PR TITLE
Support for libraryName31 and libraryName64 keys for dataServices in pluginDefinition.json

### PR DESCRIPTION
**Pull request by lchudinov**
This PR introduces support for `libraryName31` and `libraryName64` keys for dataService. The keys allow to distinguish 31 and 64-bit versions of a plugin. If bits-specific key not found then zss tries to load the dll using `libraryName` key. For example,

```json
{
  "identifier": "org.zowe.my.plugin",
  "apiVersion": "1.0.0",
  "pluginVersion": "1.2.1",
  "pluginType": "application",
  "dataServices": [
    {
      "type": "service",
      "name": "myService",
      "libraryName31": "myService31.so",
      "libraryName64": "myService64.so",
      "dependenciesIncluded": true,
      "methods": [ "GET", "POST", "DELETE" ],
      "initializerName": "myServiceServiceInstaller",
      "initializerLookupMethod": "external",
      "version": "0.0.1"
    }
  ]
}
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Zowe Common C Changelog
 
+## `1.26.0`
```
[the remark this answers is not in the document]
Comment (lchudinov) [reply CHANGELOG.md]: Done

File: c/dataservice.c [hunk 2]
```diff
@@ -143,6 +143,75 @@ static void *lookupDLLEntryPoint(char *libraryName, char *functionName){
...
+  char *libraryName = jsonObjectGetString(serviceJsonObject, LIBRARY_NAME_KEY);
+  char *initializerName = jsonObjectGetString(serviceJsonObject, INITIALIZER_NAME_KEY);
+  char dllLocation[COMMON_PATH_MAX] = {0};
+  int pluginLocationLen = strlen(pluginLocation);
+  InitializerAPI *initializer = NULL;
+
+  /* "external will load a DLL in LE on the path, and a load module in the STEPLIB in METAL */
+#ifdef METTLE
+  /* do a load of 8 char name.  Pray that the STEPLIB yields something good. */
+  zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG, "*** PANIC ***  service loading in METTLE not implemented\n");
```
[the remark this answers is not in the document]
Comment (lchudinov) [reply c/dataservice.c]: Done

File: c/dataservice.c [hunk 3]
```diff
@@ -143,6 +143,75 @@ static void *lookupDLLEntryPoint(char *libraryName, char *functionName){
...
+#else
+  if (pluginLocationLen > 0 && pluginLocation[pluginLocationLen - 1] == '/') {
+    pluginLocationLen--; /* exclude trailing slash */
+  }
+
+  do {
+    if (!libraryNameWithBits && !libraryName) {
+      zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_WARNING, "Dataservice: couldn't find '%s' or '%s' key for dataservice '%s'\n",
```
[the remark this answers is not in the document]
Comment (lchudinov) [reply c/dataservice.c:173]: `zowe-common-c` is a library and doesn't have message IDs.

File: c/dataservice.c [hunk 4]
```diff
@@ -143,6 +143,75 @@ static void *lookupDLLEntryPoint(char *libraryName, char *functionName){
...
+        LIBRARY_NAME_WITH_BITS_KEY, LIBRARY_NAME_KEY, service->identifier);
+      break;
+    }
+    if (!initializerName) {
+      zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_WARNING, "Dataservice: couldn't find '%s' key for dataservice '%s'\n",
```
[the remark this answers is not in the document]
Comment (lchudinov) [reply c/dataservice.c:178]: `zowe-common-c` is a library so it doesn't have message IDs.

**Review by 1000TurquoisePogs — APPROVED** [review]

**Pull request merged**